be/src/column/column.h

```diff
@@ -150,6 +150,9 @@ class Column {
         return append_selective(src, indexes.data(), 0, indexes.size());
     }
 
+    // Append rows from permutation
+    virtual void append_permutation(const Columns& columns, const Permutation& perm) = 0;
```

Comment:
Permutation is tight coupling with sort operator, so I think would better use visitor here.

Reply:
done